gson/src/test/java/com/google/gson/functional/ToNumberPolicyFunctionalTest.java

```diff
@@ -103,16 +104,16 @@ public void testAsListOfLongsOrDoubles() {
         .setObjectToNumberStrategy(ToNumberPolicy.LONG_OR_DOUBLE)
         .setNumberToNumberStrategy(ToNumberPolicy.LONG_OR_DOUBLE)
         .create();
-    List<Object> expected = new LinkedList<>();
+    List<Object> expected = new ArrayList<>();
     expected.add(null);
     expected.add(10L);
     expected.add(10.0);
     Type objectCollectionType = new TypeToken<Collection<Object>>() { }.getType();
     Collection<Object> objects = gson.fromJson("[null,10,10.0]", objectCollectionType);
-    assertThat(objects).isEqualTo(expected);
+    assertThat(objects).containsExactlyElementsIn(expected);
```

Comment:
`containsExactlyElementsIn` does not seem to check order (see [documentation](https://javadoc.io/doc/com.google.truth/truth/latest/com/google/common/truth/IterableSubject.html#containsExactlyElementsIn(java.lang.Iterable))). Is that intended here? If not then maybe this should continue using `isEqualTo`.

(same for the change a few lines below)

Reply:
Should tack on `.inOrder()`. Also I think we can dispense with `expected`:

```java
assertThat(objects).containsExactly(null, 10L, 10.0).inOrder();
```